packages/terminal-external/src/electron-browser/terminal-external-preference.ts

```diff
@@ -0,0 +1,101 @@
+import { PreferenceSchemaProvider } from '@theia/core/lib/browser/preferences/preference-contribution';
+import { isWindows, isOSX } from '@theia/core/lib/common/os';
+import { TerminalExternalService, TerminalExternalConfiguration } from '../common/terminal-external';
+
+export const TerminalExternalPreferences = Symbol('TerminalExternalPreferences');
+export type TerminalExternalPreferences = PreferenceProxy<TerminalExternalConfiguration>;
+
+export const TerminalExternalSchemaPromise = Symbol('TerminalExternalSchemaPromise');
+export type TerminalExternalSchemaPromise = Promise<PreferenceSchema>;
+
+export function bindTerminalExternalPreferences(bind: interfaces.Bind): void {
+    bind(TerminalExternalSchemaPromise).toDynamicValue(
+        ctx => getTerminalExternalSchema(ctx.container.get(TerminalExternalService))
+    ).inSingletonScope();
+    bind(TerminalExternalPreferences).toDynamicValue(
+        ctx => createPreferenceProxy(
+            ctx.container.get(PreferenceService),
+            ctx.container.get(TerminalExternalSchemaPromise),
+        )
+    ).inSingletonScope();
+    bind(TerminalExternalPreferenceService).toSelf().inSingletonScope();
+}
+
+@injectable()
+export class TerminalExternalPreferenceService {
+
+    @inject(TerminalExternalPreferences)
+    private readonly preferences: TerminalExternalPreferences;
+
+    @inject(PreferenceSchemaProvider)
+    private readonly preferenceSchemaProvider: PreferenceSchemaProvider;
+
+    @inject(TerminalExternalSchemaPromise)
+    private readonly promisedSchema: TerminalExternalSchemaPromise;
+
+    @postConstruct()
+    protected init(): void {
+        this.promisedSchema.then(schema => this.preferenceSchemaProvider.setSchema(schema));
+    }
+
+    /**
+     * Get the external terminal configurations from preferences.
+     */
+    getTerminalExternalConfiguration(): TerminalExternalConfiguration {
+        return {
+            'terminal.external.linuxExec': this.preferences['terminal.external.linuxExec'],
+            'terminal.external.osxExec': this.preferences['terminal.external.osxExec'],
+            'terminal.external.windowsExec': this.preferences['terminal.external.windowsExec'],
+        };
+    }
+}
+
+/**
+ * Some of the schema's properties are fetched from the backend.
+ */
+export async function getTerminalExternalSchema(terminalExternalService: TerminalExternalService): Promise<PreferenceSchema> {
+    const hostExec = await terminalExternalService.getDefaultExec();
+    return {
+        type: 'object',
+        properties: {
+            'terminal.external.windowsExec': {
+                type: 'string',
+                description: 'Customizes which terminal to run on Windows.',
+                default: `${isWindows ? hostExec : 'C:\\WINDOWS\\System32\\cmd.exe'}`
+            },
+            'terminal.external.osxExec': {
+                type: 'string',
+                description: 'Customizes which terminal application to run on macOS.',
```

Comment:
Initially, I make it this way because the opener accepts an app name as its argument (hence, a `.app` suffix for convention, i.e: `Terminal.app` or `iTerm.app`), and also because it's the same descriptions from VSCode. 

However, I aligned it with the other descriptions since I agree it isn't necessary to make a distinction here (either `Terminal` or `Terminal.app` work fine).